crates/attestation/src/crypto/crl.rs

```diff
@@ -0,0 +1,519 @@
+// SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
+// SPDX-License-Identifier: AGPL-3.0-only
+
+use crate::crypto::{SubjectPublicKeyInfoExt, TbsCertificateExt};
+
+use anyhow::{bail, ensure, Result};
+use const_oid::db::DB;
+use const_oid::ObjectIdentifier;
+use der::asn1::UIntRef;
+use der::{Encode, Sequence};
+use x509::crl::CertificateList;
+use x509::name::Name;
+use x509::PkiPath;
+
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+pub struct CrlListEntry<'a> {
+    pub url: String,
+    pub crl: CertificateList<'a>,
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Sequence)]
+pub struct CrlList<'a> {
+    pub crls: Vec<CrlListEntry<'a>>,
+}
+
+impl<'a> CrlList<'a> {
+    #[allow(unused)]
+    fn get_crl_by_url(&self, url: &str) -> Option<&CertificateList> {
+        for pair in &self.crls {
+            if pair.url == url {
+                return Some(&pair.crl);
+            }
+        }
+        None
+    }
+
+    #[allow(unused)]
+    fn get_crl_by_issuer(&self, issuer: &Name) -> Option<&CertificateList> {
+        for pair in &self.crls {
+            if &pair.crl.tbs_cert_list.issuer == issuer {
+                println!(
+                    "Returning CRL {} for issuer {}",
+                    pair.crl.tbs_cert_list.issuer, issuer
+                );
+                return Some(&pair.crl);
+            }
+        }
+        None
+    }
+
+    #[allow(unused)]
+    fn is_revoked(&self, issuer: &Name, serial: &UIntRef) -> anyhow::Result<bool> {
```

Comment:
I think that having this method will make people try to use a CRL list without validating them.
Maybe, instead, make the `issuer` a certificate, and find the correct CRL, check the signature, and then check the serial?
Alternatively, if you don't want to offer this in general, remove this method for now.

Reply:
Removed.